src/main/java/net/dries007/tfc/common/blockentities/NestBoxBlockEntity.java

```diff
@@ -42,17 +46,26 @@ public static void serverTick(Level level, BlockPos pos, BlockState state, NestB
             Entity sitter = Seat.getSittingEntity(level, pos);
             if (sitter instanceof OviparousAnimal bird)
             {
-                if (bird.isReadyForAnimalProduct())
+                if (bird.isReadyForAnimalProduct() && !nest.isFull())
                 {
                     if (bird.getRandom().nextInt(7) == 0)
                     {
-                        Helpers.playSound(level, pos, SoundEvents.CHICKEN_EGG);
-                        if (Helpers.insertOne(level, pos, TFCBlockEntities.NEST_BOX.get(), bird.makeEgg()))
+                        var event = bird.makeEggEvent();
```

Comment:
We prefer not to use `var` outside of situations with very verbose object names.

We prefer `final` before final local variables